experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -316,6 +317,49 @@ describe("SharedTree", () => {
 		});
 	});
 
+	it("Uses handles in the summary when there are no schema changes since the last summary.", async () => {
+		const provider = await TestTreeProvider.create(1, SummarizeType.onDemand);
+
+		await provider.ensureSynchronized();
+		const tree1 = provider.trees[0].schematizeInternal({
+			schema: stringSequenceRootSchema,
+			allowedSchemaModifications: AllowedUpdateType.None,
+			initialTree: ["B"],
+		});
+
+		await provider.ensureSynchronized();
+		await provider.summarize();
+
+		tree1.editableTree.insertAt(0, ["A"]);
+
+		await provider.ensureSynchronized();
+		const { summaryTree } = await provider.summarize();
+		assert(
```

Comment:
I don't think asserting aspects of the summary tree are very important for tests aimed mostly at schema index, if we care to test this it would make more sense to be in some top-level summary file. But the coverage we have from e.g. snapshot tests already prevents most regressions I can think of in this realm.

Reply:
I tried to remove them, but I can't seem to access the next level of `.tree` in the summary object without first asserting that each level is of type `SummaryType.Tree`.

However, I did move it out to a separate function to remove some duplicated code